src/libraries/Microsoft.Extensions.Configuration.Binder/gen/ConfigurationBindingGenerator.cs

```diff
@@ -30,39 +33,61 @@ public void Initialize(IncrementalGeneratorInitializationContext context)
                         ? new CompilationData((CSharpCompilation)compilation)
                         : null);
 
-            IncrementalValuesProvider<BinderInvocation?> inputCalls = context.SyntaxProvider
+            IncrementalValueProvider<(SourceGenerationSpec?, ImmutableEquatableArray<DiagnosticInfo>?)> genSpec = context.SyntaxProvider
                 .CreateSyntaxProvider(
                     (node, _) => BinderInvocation.IsCandidateSyntaxNode(node),
                     BinderInvocation.Create)
```

Comment:
Yes this can be refactored so that we can return an incremental value at each pipeline stage. Seems like this would only be for "purity": 

> so the following steps will never be able to be cut off.

I'm not sure how we can have meaningful intermediate phases with this incremental pipeline.. we still need to pass the entire object graph for each target config type per invocation.